Cold backup script fixes (#5623)
strimzi/strimzi-kafka-operator: tools/cold-backup/README.md

## Strimzi backup

> Bash script for cold/offline backups of Kafka clusters on Kubernetes/OpenShift. Only local file system is supported. Make sure to have enough free space in the target directory.

Reviewer: ```suggestion
Bash script for offline (_cold_) backups of Kafka clusters on Kubernetes or OpenShift. The script only supports a local file system. Make sure you have enough free space in the target directory.
```

Author: ok

> If you think you do not need a backup strategy for Kafka because of its embedded data replication, then consider the impact of a misconfiguration, bug or security breach that deleted all your data. For hot/online backups, you can use storage snapshotting or streaming into object storage.

Reviewer: ```suggestion
If you think you don't need a backup strategy for Kafka because of its embedded data replication, then consider the impact of a misconfiguration, bug, or security breach that deletes all of your data. For online (_hot_) backups, you can use storage snapshotting or stream into object storage.
```

Author: ok

> To run the script, the Kubernetes user must have permission to work with PVC and Strimzi custom resources. The procedure will stop the Cluster Operator and selected cluster for the duration of the backup. Before restoring the Kafka cluster you need to make sure to have the right version of Strimzi CRDs installed. If you have a single cluster-wide Cluster Operator, then you need to scale it down manually. You can run backup and restore procedures for different Kafka clusters in parallel. Only the local file system is supported. Consumer group offsets are included, but not Kafka Connect, MirrorMaker and Kafka Bridge custom resources.

Reviewer: ```suggestion
Run the script as a Kubernetes user with permission to work with PVC and Strimzi custom resources. 

The procedure will stop the Cluster Operator and a specified Kafka cluster for the duration of the backup. Before restoring the Kafka cluster, you need to make sure to have the right version of Strimzi CRDs installed. If you have a single cluster-wide Cluster Operator, then you need to scale it down manually. You can run backup and restore procedures for different Kafka clusters in parallel. Consumer group offsets are included, but not Kafka Connect, MirrorMaker and Kafka Bridge custom resources.
```

Author: ok

# Strimzi backup

Bash script for cold/offline backups of Kafka clusters on Kubernetes/OpenShift.
Only local file system is supported. Make sure to have enough free space in the target directory.

If you think you do not need a backup strategy for Kafka because of its embedded data replication, then consider the 
impact of a misconfiguration, bug or security breach that deleted all your data. For hot/online backups, you can use 
storage snapshotting or streaming into object storage.

To run the script, the Kubernetes user must have permission to work with PVC and Strimzi custom resources. The procedure 
will stop the Cluster Operator and selected cluster for the duration of the backup. Before restoring the Kafka cluster 
you need to make sure to have the right version of Strimzi CRDs installed. If you have a single cluster-wide Cluster 
Operator, then you need to scale it down manually. You can run backup and restore procedures for different Kafka 
clusters in parallel. Consumer group offsets are included, but not Kafka Connect, MirrorMaker and Kafka Bridge custom 

> resources.

Reviewer 2: If you use sentence per line as we do in most other document files, you will avoid weird re-formatings like this which just make it harder to review.

Author: Ok, will do. Anyway, I just added this phrase: Make sure to have enough free space in the target directory.

- enough disk space

## Test procedure

```sh

> CLIENT_IMAGE="quay.io/strimzi/kafka:latest-kafka-2.8.0"

Reviewer 2: Should this used the `latest` image? Wouldn't it be better to use some released image which will not change?

Author: Yep.

> krun() { kubectl run client -it --rm=true --restart=Never --image=$CLIENT_IMAGE -- $@; }

Reviewer 2: I'm not sure I like this. This probably makes it much harder to understand what the test procedure does. I also don't understand why the image name should be defined in environment variable instead of directly in the field. `krun` is AFAIK also an alias many people already have for `kubectl run`, so it might conflict.

Author: Ok. I can revert that.
